azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -583,6 +605,24 @@ func resourceArmVirtualMachineScaleSet() *schema.Resource {
 			},
 
 			"tags": tagsSchema(),
+
+			"vm_hostnames": {
```

Comment:
Where would you have them?  Those are needed when not using the auto scaling feature of the scaleSet; therefore specifying a predefined number of VMs.  We need to run remote-exec provisioner on the resulting VMs so the IP are important to us, as well as the hostnames since we pass the terraform output to a secondary script which consumes the hostnames. 

Should we put this in a Datasource instead ?

Reply:
@katbyte do you know if a data source can reference a resource that Terraform is creating ? or it only supports preexisting resources not managed by terraform?

Reply:
@djsly, a datasource can reference anything you would like, terraform created or not. However you just have to make sure the graph knows about the dependancy by linking back to the terraform resource for the name of the azure resource to load. 

With respect to these additional attributes we please remove them from this PR so I can review it one final time before merging? That really is separate task so it would be great if you could open an issue regarding the datasource so my colleagues can weigh in 🙂 

Thanks!